lib/screens/v2/candidate_generator/elevator/closures.ex

```diff
@@ -0,0 +1,160 @@
+          app_params: %Elevator{
+            elevator_id: elevator_id
+          }
+        },
+        now \\ DateTime.utc_now()
+      ) do
+    with {:ok, %Stop{id: stop_id}} <- @facility.fetch_stop_for_facility(elevator_id),
+         {:ok, location_context} <- @stop.fetch_location_context(Elevator, stop_id, now),
+         {:ok, parent_station_map} <- @stop.fetch_parent_station_name_map(),
+         {:ok, alerts} <- @alert.fetch_elevator_alerts_with_facilities() do
+      elevator_closures = relevant_alerts(alerts)
+      routes_map = get_routes_map(elevator_closures, stop_id)
+
+      {in_station_alerts, outside_alerts} =
+        split_alerts_by_location(elevator_closures, location_context)
+
+      [
+        %ElevatorClosures{
+          id: elevator_id,
+          in_station_alerts: Enum.map(in_station_alerts, &alert_to_elevator_closure/1),
+          other_stations_with_alerts:
+            format_outside_alerts(outside_alerts, parent_station_map, routes_map)
+        }
+      ]
+    else
+      :error ->
+        []
+
+      {:error, error} ->
+        Logger.error("[elevator_status_instances] #{inspect(error)}")
+        []
+    end
+  end
+
+  defp relevant_alerts(alerts) do
+    Enum.filter(alerts, &(&1.effect == :elevator_closure))
+  end
+
+  defp get_routes_map(elevator_closures, home_parent_station_id) do
+    elevator_closures
+    |> get_parent_station_ids_from_entities()
+    |> MapSet.new()
+    |> MapSet.put(home_parent_station_id)
+    |> Enum.map(fn station_id ->
+      {station_id, station_id |> route_ids_serving_stop() |> routes_to_labels()}
+    end)
+    |> Enum.into(%{})
+  end
+
+  defp get_parent_station_ids_from_entities(alerts) do
+    alerts
+    |> Enum.flat_map(fn %Alert{informed_entities: informed_entities} ->
+      informed_entities
+      |> Enum.map(fn %{stop: stop_id} -> stop_id end)
+      |> Enum.filter(&String.starts_with?(&1, "place-"))
+    end)
+  end
+
+  defp route_ids_serving_stop(stop_id) do
+    case @route.fetch(%{stop_id: stop_id}) do
+      {:ok, routes} -> routes
+      # Show no route pills instead of crashing the screen
+      :error -> []
+    end
+  end
+
+  defp routes_to_labels(routes) do
+    routes
+    |> Enum.map(fn
+      %Route{type: :subway, id: id} -> id |> String.downcase() |> String.to_atom()
+      %Route{type: :light_rail, id: "Green-" <> _} -> :green
+      %Route{type: :light_rail, id: "Mattapan" <> _} -> :mattapan
+      %Route{type: :bus, short_name: "SL" <> _} -> :silver
+      %Route{type: :rail} -> :cr
+      %Route{type: type} -> type
+    end)
+    |> Enum.uniq()
+  end
+
+  defp split_alerts_by_location(alerts, location_context) do
+    Enum.split_with(alerts, fn %Alert{informed_entities: informed_entities} ->
+      location_context.home_stop in Enum.map(informed_entities, & &1.stop)
+    end)
+  end
+
+  defp get_informed_facility(entities) do
+    entities
+    |> Enum.find_value(fn
+      %{facility: facility} -> facility
+      _ -> false
```

Comment:
* The place where this is used in `alert_to_elevator_closure` doesn't seem to account for the possibility that it could be `nil`. Ultimately this could happen if there were an alert that impacted the `USING_WHEELCHAIR` activity but did not inform any facilities. We should explicitly handle that possibility in some way, whether it's to ignore/filter out that case, or crash on it.
* What happens if an alert affects multiple facilities? Again, maybe we could either explicitly ignore or crash.

Reply:
No harm in handling them, but I think in both cases that would be a data issue. With that in mind, I think I'll just filter them out.